src/utils/form-actions.ts

```diff
@@ -78,3 +81,27 @@ export async function sendEmail(
     });
   }
 }
+
+export async function getTemplates(): Promise<Template[] | []> {
+  const { data, errors } =
+    await getAmplifyBackendClient().models.TemplateStorage.list();
+
+  if (errors) {
+    logger.error('Failed to get templates', errors);
+  }
+
+  if (!data) {
+    logger.warn(`Failed to retrieve templates`);
+    return [];
+  }
+
+  const parsedData: Template[] = data
+    .map((template) => {
+      if (isTemplateValid(template)) {
+        return $Template.parse(template);
```

Comment:
I'm pretty sure `$Template.parse` will throw an error if the validation fails. Which is why we've been using `safeParse`. I could be wrong though, might be worth a check.

Reply:
Oh I just saw you're doing a `.safeParse` in `isTemplateValid`. Why not just return data from the function? Instead of doing a `parse` twice?

Reply:
`safeParse` also returns the validated data too. So doing the additional `parse` seems a little redundant. Feels like you might as well return the data  or undefined from `isTemplateValid`.

So

```
export const isTemplateValid = (
  input: unknown
): input is z.infer<typeof $Template | undefined> => $Template.safeParse(input).data;
```

This map already filters out undefined values. 

So you can they just do

```
.map(isTemplateValid).filter((template): template is Template => template !== undefined)
```

Reply:
Fixed now